src/System.IO/src/System/IO/StreamReader.cs

```diff
@@ -526,7 +526,10 @@ private void DetectEncoding()
                 _byteBuffer[2] == 0xFE && _byteBuffer[3] == 0xFF)
             {
                 // Big Endian UTF32
-                _encoding = new UTF32Encoding(bigEndian: true, byteOrderMark: true);
+                // Unfortunately, there is no public Encoding.BigEndianUTF32 property, so instead
+                // we call Encoding.GetEncoding, which will retrieve a cached instance.
+                const int CodePageUTF32BE = 12001;
+                _encoding = Encoding.GetEncoding(CodePageUTF32BE);
```

Comment:
> My point is, if we're concerned about this change, why wasn't there a concern about existing uses of Encoding.GetEncoding("utf-8"), Encoding.GetEncoding("utf-16"), etc. (an API available since .NET 1.1) when the EncodingProvider feature was added in .NET 4.6, which could change the behavior in the same way you suggest if someone registers "utf-8", "utf-16", etc. to return different instances?

when added the Provider, there was any comparability concerns because apps didn't have providers before. for the new apps, the app decide to opt-in using the provider and they should be understanding this can give different behaviors for encoding everywhere in the app (networking, system.io...etc.). it is the app decision to change the behavior.